components/entitlement/org.wso2.carbon.identity.entitlement.endpoint/src/main/java/org/wso2/carbon/identity/entitlement/endpoint/util/JSONRequestParser.java

```diff
@@ -105,113 +105,124 @@ public static RequestCtx parse(String jsonRequest) throws JsonParseException, Re
-                        }
-                        if (jsonCategory.has(EntitlementEndpointConstants.ID)) {
-                            id = jsonCategory.get(EntitlementEndpointConstants.ID).getAsString();
-                        }
-                        if (jsonCategory.has(EntitlementEndpointConstants.CONTENT)) {
-                            ByteArrayInputStream inputStream;
-                            DocumentBuilderFactory dbf;
-                            Document doc = null;
-
-                            String xmlContent = stringContentToXMLContent(jsonCategory
-                                    .get(EntitlementEndpointConstants.CONTENT)
-                                    .getAsString());
-                            inputStream = new ByteArrayInputStream(xmlContent.getBytes());
-                            dbf = IdentityUtil.getSecuredDocumentBuilderFactory();
-                            dbf.setNamespaceAware(true);
-
-                            try {
-                                doc = dbf.newDocumentBuilder().parse(inputStream);
-                            } catch (Exception e) {
-                                throw new JsonParseException("DOM of request element can not be created from String");
-                            } finally {
-                                try {
-                                    inputStream.close();
-                                } catch (IOException e) {
-                                    throw new JsonParseException("DOM of request element can not be created from String");
-                                }
-                            }
-                            if (doc != null) {
-                                content = doc;
-                            }
-                        }
+        RequestCtx requestCtx = new RequestCtx(null, categories, returnPolicyIdList, combinedDecision,
+                multiRequests, requestDefaults);
+        return requestCtx;
+    }
 
-                        //add all category attributes
-                        if (jsonCategory.has(EntitlementEndpointConstants.ATTRIBUTE)) {
-                            if (jsonCategory.get(EntitlementEndpointConstants.ATTRIBUTE).isJsonArray()) {
-                                attributes = new HashSet<>();
-                                for (JsonElement jsonElement : jsonCategory.get(EntitlementEndpointConstants.ATTRIBUTE)
-                                        .getAsJsonArray()) {
-                                    attributes.add(jsonObjectToAttribute(jsonElement.getAsJsonObject()));
+    private static void jsonAttributeSeperator(Map.Entry<String, JsonElement> jsonAttribute, JsonObject jsonCategory, Set<Attributes> categories) throws RequestParseException, UnknownIdentifierException {
+
+        Node content = null;
+        URI category = null;
+        Set<Attribute> attributes = null;
+        String id = null;
+        switch (jsonAttribute.getKey()) {
+
+            //For a custom Category
```

Comment:
Add a space before starting the comment. 
Is there any specific reason this comment into two lines?